src/storage/snapshot.rs

```diff
@@ -0,0 +1,161 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use tokio::sync::RwLock;
+use wasmcloud_control_interface::LinkDefinition;
+
+use crate::storage::{Actor, Host, Provider, ReadStore, StateKind};
+use crate::workers::LinkSource;
+
+// NOTE(thomastaylor312): This type is real ugly and we should probably find a better way to
+// structure the ReadStore trait so it doesn't have the generic T we have to work around here. This
+// is essentially a map of "state kind" -> map of ID to partially serialized state. I did try to
+// implement some sort of getter trait but it has to be generic across T
+type InMemoryData = HashMap<String, HashMap<String, serde_json::Value>>;
+
+/// A store and claims/links source implementation that contains a static snapshot of the data that
+/// can be refreshed periodically. Please note that this is scoped to a specific lattice ID and
+/// should be constructed separately for each lattice ID.
+///
+/// NOTE: This is a temporary workaround until we get a proper caching store in place
+pub struct SnapshotStore<S, L> {
+    store: S,
+    link_source: L,
+    lattice_id: String,
+    stored_state: Arc<RwLock<InMemoryData>>,
+    links: Arc<RwLock<Vec<LinkDefinition>>>,
+}
+
+impl<S, L> Clone for SnapshotStore<S, L>
+where
+    S: Clone,
+    L: Clone,
+{
+    fn clone(&self) -> Self {
+        Self {
+            store: self.store.clone(),
+            link_source: self.link_source.clone(),
+            lattice_id: self.lattice_id.clone(),
+            stored_state: self.stored_state.clone(),
+            links: self.links.clone(),
+        }
+    }
+}
```

Comment:
Do you need this to restrict clone to where S, L are clonable?

Reply:
To clarify, I think this code is correct just asking if that's the reason why this isn't `derive(Clone)`

Reply:
This is for cleanliness here mostly. We know that anything we pass in here will be clone, but many times you can’t actually know that. For us here, if we want to derive clone, we’d have to add a clone bound to the top level strict definition. Then that clone bound bleeds to every other impl of the struct